Common/Python/PythonInitializer.cs

```diff
@@ -111,11 +121,29 @@ public static bool AddPythonPaths(IEnumerable<string> paths)
                     var currentPath = pythonCurrentPath.As<List<string>>();
                     _pendingPathAdditions = _pendingPathAdditions.Where(x => !currentPath.Contains(x.Replace('\\', '/'))).ToList();
 
+                    // Algorithm location most always be before any other path added through this method
+                    var insertionIndex = 0;
+                    if (!_algorithmLocation.IsNullOrEmpty())
+                    {
+                        if (currentPath.Contains(_algorithmLocation.Replace('\\', '/')))
+                        {
+                            // The algorithm location is already there. Let's add the other paths after it
+                            insertionIndex = currentPath.IndexOf(_algorithmLocation.Replace('\\', '/')) + 1;
```

Comment:
> currentPath.IndexOf(_algorithmLocation.Replace('\\', '/'))

Can directly call this guy and skip the contains above? will return -1 if not found I believe